src/DurableSDK/OrchestrationInvoker.cs

```diff
@@ -11,58 +11,98 @@ namespace Microsoft.Azure.Functions.PowerShellWorker.Durable
 
-                var (shouldStop, actions) =
-                    orchestrationBindingInfo.Context.OrchestrationActionCollector.WaitForActions(asyncResult.AsyncWaitHandle);
+            var asyncResult = powerShellServices.BeginInvoke(outputBuffer);
 
-                if (shouldStop)
+            var (shouldStop, actions) =
+                orchestrationBindingInfo.Context.OrchestrationActionCollector.WaitForActions(asyncResult.AsyncWaitHandle);
+
+            if (shouldStop)
+            {
+                // The orchestration function should be stopped and restarted
+                powerShellServices.StopInvoke();
+                // return (Hashtable)orchestrationBindingInfo.Context.OrchestrationActionCollector.output;
+                return CreateOrchestrationResult(isDone: false, actions, output: null, context.CustomStatus);
+            }
+            else
+            {
+                try
                 {
-                    // The orchestration function should be stopped and restarted
-                    pwsh.StopInvoke();
-                    return CreateOrchestrationResult(isDone: false, actions, output: null, context.CustomStatus);
+                    // The orchestration function completed
+                    powerShellServices.EndInvoke(asyncResult);
+                    var result = CreateReturnValueFromFunctionOutput(outputBuffer);
+                    return CreateOrchestrationResult(isDone: true, actions, output: result, context.CustomStatus);
                 }
-                else
+                catch (Exception e)
                 {
-                    try
-                    {
-                        // The orchestration function completed
-                        pwsh.EndInvoke(asyncResult);
-                        var result = FunctionReturnValueBuilder.CreateReturnValueFromFunctionOutput(outputBuffer);
-                        return CreateOrchestrationResult(isDone: true, actions, output: result, context.CustomStatus);
-                    }
-                    catch (Exception e)
-                    {
-                        // The orchestrator code has thrown an unhandled exception:
-                        // this should be treated as an entire orchestration failure
-                        throw new OrchestrationFailureException(actions, context.CustomStatus, e);
-                    }
+                    // The orchestrator code has thrown an unhandled exception:
+                    // this should be treated as an entire orchestration failure
+                    throw new OrchestrationFailureException(actions, context.CustomStatus, e);
                 }
             }
-            finally
+        }
+
+        public static object CreateReturnValueFromFunctionOutput(IList<object> pipelineItems)
+        {
+            if (pipelineItems == null || pipelineItems.Count <= 0)
             {
-                pwsh.ClearStreamsAndCommands();
+                return null;
```

Comment:
Just for clarity: the `pwsh.ClearStreamsAndCommand()` call here didn't use to be  in the `CreateReturnValueFromFunctionOutput` method. The implementation of this method has not changed at all, it's just that the diff is acting up and still showing code from the previous `Invoke` implementation.